source/server/server.cc

```diff
@@ -64,6 +63,19 @@
 
 namespace Envoy {
 namespace Server {
+namespace {
+std::unique_ptr<ConnectionHandler> getHandler(Event::Dispatcher& dispatcher) {
+
+  auto* factory = Config::Utility::getFactoryByName<ConnectionHandlerFactory>(
+      "envoy.connection_handler.default");
+  if (factory) {
+    return factory->createConnectionHandler(dispatcher, absl::nullopt);
+  }
+  ENVOY_LOG_MISC(debug, "Unable to find envoy.connection_handler.default factory");
```

Comment:
Why not raise an exception? if returns nullptr, it will crash Envoy.

Reply:
this should never happen in upstream Envoy.  it can in Envoy mobile if you misconfigure things.

Reply:
Would that be due to failing to register the factory? Or is there some config that drives this?

In general EM does crash on missing registration, though I guess just logging might be better?

Reply:
This is hit in Envoy Mobile which makes mobile application crash after we call `addDirectResponse` on `TestEngineBuilder` (as it adds a listener) (iOS only API).